pyqtgraph/graphicsItems/PlotCurveItem.py

```diff
@@ -379,7 +387,7 @@ def updateData(self, *args, **kargs):
         if 'stepMode' in kargs:
             self.opts['stepMode'] = kargs['stepMode']
 
-        if self.opts['stepMode'] is True:
+        if self.opts['stepMode'] in ("center", True):  ## check against True for backwards compatibility
             if len(self.xData) != len(self.yData)+1:  ## allow difference of 1 for step mode plots
```

Comment:
I would also consider adding a `DeprecationWarning` if `stepMode is True`

Reply:
2335e29 does it.


Note 1: I did the check on the `PlotDataItem.setData()` and `PlotCurveItem.updateData()` instead of the constructors, to cover the possibility of those being called directly. If stepMode is used in the constructors, these methods end up being called anyway.

Note 2: since the `warnings` module is not otherwise used in these modules, I chose to import it locally so that the whole deprecation check can be cleanly removed once it is no longer needed.

Reply:
Thanks for pointing this out; this looks good to me.